arch/x86/kvm/pkvm/pkvm.h

```diff
@@ -8,6 +8,41 @@
 DECLARE_PER_CPU(struct pkvm_pcpu *, phys_cpu);
 DECLARE_PER_CPU(struct kvm_vcpu *, host_vcpu);
 
+/* Represents a guest VM. */
+struct pkvm_vm {
+	/* Point to the kvm structure owned by the host */
+	struct kvm *shared_kvm;
+	/* Structure size */
+	size_t size;
+	/*
+	 * The struct kvm should be the last element. In cases where struct kvm
+	 * is wrapped by a vendor specific structure, putting it as the last
+	 * element can safely extend the size of pkvm_vm w/o affecting layout
+	 * compatibility.
+	 *
+	 * If struct kvm is wrapped by a vendor specific structure, it must reside
+	 * at offset 0 of that structure. This ensures that &pkvm_vm->kvm correctly
+	 * resolves to the underlying struct kvm instance. It is the responsibility
+	 * of vendor code to guarantee this layout.
+	 */
+	struct kvm kvm;
+};
+
+/* The pkvm_vm structure size w/o struct kvm */
+#define PKVM_VM_BASE_SIZE		(offsetof(struct pkvm_vm, kvm))
+
+static inline struct pkvm_vm *to_pkvm(struct kvm *kvm)
+{
+	/*
+	 * Compiling check to guarantee the kvm structure is the last element of
```

Comment:
nit: It is obvious that BUILD_BUG_ON is a compiling check, so the comment could be shorter, e.g. "Make sure the kvm structure is the last element of pkvm_vm. See comments of struct pkvm_vm." ?